File: Week1/deaf-grandma/PY/deaf_grandma.py

def deaf_grandma():
    bye_count = 0
    
    while True:
        input_text = input('Kid: ')
        if input_text == '':
            print('WHAT?')
        elif not input_text.isupper():
            print('SPEAK UP, KID!')
        elif input_text == "GOODBYE!":
            bye_count  = bye_count + 1
            if bye_count <= 1:
                print('LEAVING SO SOON?')
            elif bye_count >= 2:
                print('LATER SKATER!')
                break
        elif input_text.isupper():
            print('NO, NOT SINCE 1946!')
        
deaf_grandma()

File: Week1/deaf-grandma/RB/deaf_grandma.rb
def deaf_grandma()
  bye_count = 0

  while true
    puts 'Kid: '
    input_text = gets.chomp
    if input_text == ''
      puts 'Grandma:'    
      puts 'WHAT?'
    elsif input_text != input_text.upcase
      puts 'Grandma:'    
      puts 'SPEAK UP, KID!'    
    elsif input_text == 'GOODBYE!'
      bye_count += 1
      if bye_count <= 1
        puts 'Grandma:'    
        puts 'LEAVING SO SOON?'
      elsif bye_count >= 2
        puts 'Grandma:'
        puts 'LATER SKATER'
        break
      end
    elsif input_text == input_text.upcase
      puts 'Grandma:'
      puts 'NO, NOT SINCE 1946!'
    end
  end
end

deaf_grandma()